daemon/cmd/daemon_main.go

```diff
@@ -1183,7 +1188,20 @@ func initEnv(cmd *cobra.Command) {
 	// Configure k8s as soon as possible so that k8s.IsEnabled() has the right
 	// behavior.
 	bootstrapStats.k8sInit.Start()
-	k8s.Configure(option.Config.K8sAPIServer, option.Config.K8sKubeConfigPath, defaults.K8sClientQPSLimit, defaults.K8sClientBurst)
+	// The flag K8sAPIServerURLs takes precedence over K8sAPIServer.
```

Comment:
Given the old flag is deprecated instead of mentioning precedence order in flag help we can log a warning if the user sets both the options as you mentioned.